controllers/cassandrabackup_controller.go

```diff
@@ -80,20 +80,41 @@ func (r *CassandraBackupReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 		}
 	}
 
+	// If there is anything in progress, simply requeue the request
+	if len(backup.Status.InProgress) > 0 {
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
 	if backup.GetDeletionTimestamp() != nil {
 		if ctrlutil.ContainsFinalizer(backup, finalizerName) {
-			// Backup is being deleted, call Medusa to remove it also from storage
-			err = r.removeBackup(ctx, backup)
-			if err != nil {
-				return ctrl.Result{RequeueAfter: 10 * time.Second}, err
-			}
+			// Deletion process has already been started, return later
+			patch := client.MergeFrom(backup.DeepCopy())
+			backup.Status.InProgress = append(backup.Status.InProgress, "removeBackup")
```

Comment:
I am not a fan of overloading the `InProgress` property. I would prefer a separate property if necessary. 

The question is, do we need another property? I think the answer is yes, mostly to notify the client of a failure so that further remediation can be performed.

Reply:
I wanted to avoid modifying the CRD, but yes - I can add a new property.

Reply:
But what point is there for a deletion inprogress if we have deletion timestamp?

Reply:
I am not particularly thrilled about an additional status field for deletions, but I am even less thrilled about the idea of overloading one of the existing fields. 

The most important thing in my mind is that we communicate that the deletion failed.